app/services/ThrottlingService.scala

```diff
@@ -78,13 +78,13 @@ trait ThrottlingService extends BSONBuilderHelpers {
 
   def throttle(initialLocation: Location, auditContext: TAuditContext, ruleContext: RuleContext)(implicit request: Request[AnyContent], ex: ExecutionContext): Future[Location] = {
 
-    def location(credId: String) = {
+    def location(userDiscriminator: String) = {
```

Comment:
this can be the identifier type instead of string, then the flatMap on line 123 will not need the `(_)`